infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultAdapterForUpgrade.java

```diff
@@ -27,18 +30,59 @@ public String getVersion() {
                 // Just for wrap this up
             }
 
-            LogReplicationStreams.VersionString versionStringKey =
-                    LogReplicationStreams.VersionString.newBuilder()
-                            .setName("VERSION").build();
             try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
-                versionString =
-                        ((LogReplicationStreams.Version)txn.getRecord(VERSION_TEST_TABLE,
-                                versionStringKey).getPayload()).getVersion();
+                getVersion(txn);
                 txn.commit();
             } catch (Exception e) {
                 // Just for wrap this up
             }
         }
         return versionString;
     }
+
+    private String getVersion(TxnContext txn) {
+        LogReplicationStreams.VersionString versionStringKey =
+                LogReplicationStreams.VersionString.newBuilder()
+                        .setName("VERSION").build();
+        versionString =
+                ((LogReplicationStreams.Version)txn.getRecord(VERSION_TEST_TABLE,
+                        versionStringKey).getPayload()).getVersion();
+        return versionString;
+    }
+
+    /**
+     * @return false if the cluster version is already at my version
+     * and true otherwise indicating that cluster is undergoing a version upgrade
+     */
+    @Override
+    public boolean isUpgradeRunning(TxnContext txnContext) {
+        return versionString.equals(getVersion(txnContext));
+    }
+
+    public static void startRollingUpgrade(CorfuStore corfuStore) {
```

Comment:
Can you add an annotation or comment here that this method is for testing purposes only?

Reply:
done. thanks